packages/x-data-grid-premium/src/hooks/features/rowGrouping/createGroupingColDef.tsx

```diff
@@ -52,10 +52,7 @@ const GROUPING_COL_DEF_FORCED_PROPERTIES_DATA_SOURCE: Pick<
  * TODO: Make this index comparator depth invariant, the logic should not be inverted when sorting in the "desc" direction (but the current return format of `sortComparator` does not support this behavior).
  */
 const groupingFieldIndexComparator: GridComparatorFn = (v1, v2, cellParams1, cellParams2) => {
-  const model = gridRowGroupingSanitizedModelSelector(
-    cellParams1.api.state,
-    cellParams1.api.instanceId,
-  );
+  const model = gridRowGroupingSanitizedModelSelector({ current: cellParams1.api });
```

Comment:
A step towards https://github.com/mui/mui-x/issues/11440

Reply:
We're passing `api` in a few places, we should probably consider replacing those with `apiRef`.

Reply:
I think it couples more with #11440, let's consider it when working on that.